server/src/main/java/org/cloudfoundry/identity/uaa/client/ClientAdminEndpointsValidator.java

```diff
@@ -96,12 +94,12 @@ public ClientDetails validate(ClientDetails prototype, boolean create, boolean c
 
         if (requestedGrantTypes.isEmpty()) {
             throw new InvalidClientDetailsException("An authorized grant type must be provided. Must be one of: "
-                            + VALID_GRANTS.toString());
+                            + OauthGrant.SUPPORTED_GRANTS.toString());
         }
         for (String grant : requestedGrantTypes) {
-            if (!VALID_GRANTS.contains(grant)) {
+            if (!OauthGrant.SUPPORTED_GRANTS.contains(grant.toLowerCase())) {
                 throw new InvalidClientDetailsException(grant + " is not an allowed grant type. Must be one of: "
-                                + VALID_GRANTS.toString());
+                                + OauthGrant.SUPPORTED_GRANTS.toString());
```

Comment:
If using enum ...the line 100 could be replaced with ...
OauthGrant.forName(grant)==null
and 102 could be...by replacing OauthGrant.SUPPORTED_GRANTS.toString() with 
Arrays.toString(OauthGrant.values())